### PR TITLE
Added Get Endpoint for CompanyAccount and SmallFull

### PR DESCRIPTION
**Pull request by gpatrac**

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/api/accounts/controller/SmallFullController.java [hunk 1]
```diff
@@ -32,4 +33,14 @@ public ResponseEntity create(@Valid @RequestBody SmallFull smallFull,
         SmallFull result = smallFullService.save(smallFull, transaction.getCompanyNumber());
         return ResponseEntity.status(HttpStatus.CREATED).body(result);
     }
+
+    @GetMapping(value = "/transactions/{transactionId}/company-accounts/{companyAccountId}/small-full",
+        produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity getCompanyAccount(HttpServletRequest request) {
+
+        SmallFull smallFull = (SmallFull) request.getSession()
```
Comment (jsimons1989) [inline src/main/java/uk/gov/companieshouse/api/accounts/controller/SmallFullController.java]: A `SmallFullEntity` is stored in the session in the interceptor

File: src/main/java/uk/gov/companieshouse/api/accounts/controller/CompanyAccountController.java [hunk 2]
```diff
@@ -40,4 +48,15 @@ public ResponseEntity createCompanyAccount(@Valid @RequestBody CompanyAccount co
                 return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(null);
         }
     }
+
+    @GetMapping(value = "/transactions/{transactionId}/company-accounts/{companyAccountId}",
+        produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity getCompanyAccount(HttpServletRequest request) {
+
+        CompanyAccountEntity companyAccountEntity = (CompanyAccountEntity) request.getSession()
+            .getAttribute(AttributeName.COMPANY_ACCOUNT.getValue());
```
Comment (jsimons1989) [inline src/main/java/uk/gov/companieshouse/api/accounts/controller/CompanyAccountController.java]: This should check that the company account was found successfully and return a 404 if it isn't in the session.

File: src/main/java/uk/gov/companieshouse/api/accounts/controller/SmallFullController.java (hunk 1 above)
Comment (sroberts2) [reply src/main/java/uk/gov/companieshouse/api/accounts/controller/SmallFullController.java]: Good spot. I will refactor the name of the SmallFullEntity stored via the SmallFullInterceptor from SmallFull to SmallFullEntity I think that may have been responsible for the confusion.

Comment (jsimons1989) [reply src/main/java/uk/gov/companieshouse/api/accounts/controller/SmallFullController.java]: This should have failed during testing with a cast exception?

File: src/main/java/uk/gov/companieshouse/api/accounts/controller/SmallFullController.java [hunk 3]
```diff
@@ -42,9 +43,9 @@ public ResponseEntity create(@Valid @RequestBody SmallFull smallFull,
         produces = MediaType.APPLICATION_JSON_VALUE)
     public ResponseEntity getCompanyAccount(HttpServletRequest request) {
 
-        SmallFull smallFull = (SmallFull) request.getSession()
+        SmallFullEntity smallFullEntity = (SmallFullEntity) request
             .getAttribute(AttributeName.SMALLFULL.getValue());
 
-        return ResponseEntity.status(HttpStatus.OK).body(smallFull);
+        return ResponseEntity.status(HttpStatus.OK).body(smallFullEntity.getData());
```
Comment (jsimons1989) [inline src/main/java/uk/gov/companieshouse/api/accounts/controller/SmallFullController.java]: The data needs to be transformed to the model matching the spec.

File: src/main/java/uk/gov/companieshouse/api/accounts/controller/CompanyAccountController.java [hunk 4]
```diff
@@ -38,4 +51,23 @@ public ResponseEntity createCompanyAccount(@Valid @RequestBody CompanyAccount co
...
+        CompanyAccountEntity companyAccountEntity = (CompanyAccountEntity) request
+                .getAttribute(AttributeName.COMPANY_ACCOUNT.getValue());
+        if (companyAccountEntity == null) {
+            debugMap.put("message",
+                    "CompanyAccountController error: No company account in request");
+            LOGGER.errorRequest(request, null, debugMap);
+            return ResponseEntity.status(HttpServletResponse.SC_NOT_FOUND).body(null);
+        }
+
+        return ResponseEntity.status(HttpStatus.OK).body(companyAccountEntity.getData());
```
Comment (jsimons1989) [inline src/main/java/uk/gov/companieshouse/api/accounts/controller/CompanyAccountController.java]: The data needs to be transformed to the model matching the spec.

File: src/main/java/uk/gov/companieshouse/api/accounts/controller/SmallFullController.java [hunk 5]
```diff
@@ -35,4 +49,22 @@ public ResponseEntity create(@Valid @RequestBody SmallFull smallFull,
...
+    public ResponseEntity get(HttpServletRequest request) {
+
+        final Map<String, Object> debugMap = new HashMap<>();
+        debugMap.put("request_method", request.getMethod());
+
+        SmallFullEntity smallFullEntity = (SmallFullEntity) request
+                .getAttribute(AttributeName.SMALLFULL.getValue());
+        if (smallFullEntity == null) {
+            debugMap.put("message",
+                    "CompanyAccountController error: No small-full account in request");
```
Comment (jsimons1989) [inline src/main/java/uk/gov/companieshouse/api/accounts/controller/SmallFullController.java]: Wrong controller name

File: src/main/java/uk/gov/companieshouse/api/accounts/controller/SmallFullController.java [hunk 6]
```diff
@@ -35,4 +49,22 @@ public ResponseEntity create(@Valid @RequestBody SmallFull smallFull,
...
+            LOGGER.errorRequest(request, null, debugMap);
+            return ResponseEntity.status(HttpServletResponse.SC_NOT_FOUND).body(null);
+        }
+
+        return ResponseEntity.status(HttpStatus.OK).body(smallFullEntity.getData());
```
Comment (jsimons1989) [inline src/main/java/uk/gov/companieshouse/api/accounts/controller/SmallFullController.java]: The data needs to be transformed to match the spec

**Pull request merged**